crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -503,8 +540,10 @@ unsafe impl<'a, D: QueryData + 'static, F: QueryFilter + 'static> SystemParam
                 world.change_tick(),
             )
         };
-        let result = query.single_inner();
-        !matches!(result, Err(QuerySingleError::MultipleEntities(_)))
+        match query.single_inner() {
+            Ok(_) | Err(QuerySingleError::NoEntities(_)) => ValidationOutcome::Valid,
+            Err(QuerySingleError::MultipleEntities(_)) => ValidationOutcome::Skipped,
+        }
```

Comment:
this is the same as before, but why is it like that?
0 would give None
1 would give Some
n would give Skipped?

Reply:
Yeah, that was the intended design. I'm not sold on it being particularly useful.

Reply:
I recommend we continue discussion about improving the semantics of the existing system either in https://github.com/bevyengine/bevy/issues/18516 or a new issue.